src/features/unified-share-modal/UnifiedShareModal.js

```diff
@@ -486,8 +512,9 @@ class UnifiedShareModal extends React.Component<Props, State> {
                             inlineNotice={inlineNotice}
                             isContactsFieldEnabled={canInvite}
                             isExpanded={isInviteSectionExpanded}
-                            isExternalUserSelected={isExternalUserIninviteCollabsContacts}
+                            isExternalUserSelected={this.hasExternalContact(INVITE_COLLABS_CONTACTS_TYPE)}
                             onContactInput={this.openInviteCollaborators}
+                            onPillCreate={this.onPillCreate(INVITE_COLLABS_CONTACTS_TYPE)}
```

Comment:
Render is called many times, you really want to create closures per render?
Can the onPillCreate function be created once and saved on the `this` object?
Also perhaps `getOnPillCreate` might be a better name, since it denotes building a function.

Reply:
yeah, I would recommend saving the function to this and re-using that as well. I'd opt for `this.handleInviteCollabPillCreate` and `this.handleEmailSharedLinkPillCreate` or some equivalent.

Reply:
really good point